File: Codigo en 4 lenguajes/js/EjemploMetodosConParametros.js
// EjemploMetodosConParametros.js
export default class EjemploMetodosConParametros {
  mostrar() {
    console.log("=== Ejemplo: Métodos con parámetros ===");

    class Calculadora {
      sumar(a, b) { return a + b; }
      multiplicar(a, b) { return a * b; }
      calcularIVA(precio) { return precio * 0.19; }
      registrarPersona(nombre, edad) { console.log(`  Registrado: ${nombre}, ${edad} años`); }
      esMayor(edad) { return edad >= 18; }
    }

    const calc = new Calculadora();
    console.log("Sumar 2+3 =", calc.sumar(2,3));
    console.log("Multiplicar 4*5 =", calc.multiplicar(4,5));
    console.log("Calcular IVA de 100 =", calc.calcularIVA(100.0));
    console.log("Registrar persona:");
    calc.registrarPersona("María", 22);
    console.log("Verificar mayor (edad 17):", calc.esMayor(17));
    console.log();
  }
}

File: Codigo de los 4 lenguajes/C#/EjemploMetodosConParametros.cs
using System;

public class Calculadora
{
    public int Sumar(int a,int b){ return a+b; }
    public int Multiplicar(int a,int b){ return a*b; }
    public double CalcularIVA(double precio){ return precio*0.19; }
    public void RegistrarPersona(string nombre,int edad){ Console.WriteLine(" -> Registrado: " + nombre + ", " + edad + " años"); }
    public bool EsMayor(int edad){ return edad>=18; }
}

public class EjemploMetodosConParametros
{
    public void Mostrar()
    {
        Console.WriteLine(" Ejemplo: Métodos con parámetros ");
        Calculadora calc = new Calculadora();
        Console.WriteLine("Sumar 2+3 = " + calc.Sumar(2,3));
        Console.WriteLine("Multiplicar 4*5 = " + calc.Multiplicar(4,5));
        Console.WriteLine("Calcular IVA de 100 = " + calc.CalcularIVA(100.0));
        Console.WriteLine("Registrar persona: " );
        calc.RegistrarPersona("María",22);
        Console.WriteLine("Verificar mayor (edad 17): " + calc.EsMayor(17));
        Console.WriteLine();
    }
}
